Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BoxComponentMode.cpp

```diff
@@ -12,6 +12,11 @@ namespace AzToolsFramework
 {
     AZ_CLASS_ALLOCATOR_IMPL(BoxComponentMode, AZ::SystemAllocator, 0)
 
+    void BoxComponentMode::Reflect(AZ::ReflectContext* context)
+    {
+        AzToolsFramework::ComponentModeFramework::ReflectEditorBaseComponentModeDescendant<BoxComponentMode>(context);
+    }
+
```

Comment:
Does this boilerplate need to be added to *every* component mode? If so, I wonder if EditorBaseComponentMode should become a template that produces this bit automatically?

As an example, look at https://github.com/o3de/o3de/blob/development/Gems/GradientSignal/Code/Include/GradientSignal/Editor/EditorGradientComponentBase.h . It provides a Reflect() method that adds a gradient previewer to every gradient component.

Reply:
Good spot, I will check this out

Reply:
Added to the follow-up investigation to clean up identifiers